tests/test_resource_tracker.py

```diff
@@ -0,0 +1,227 @@
+            name1 = f.readline().rstrip().decode('ascii')
+            name2 = f.readline().rstrip().decode('ascii')
+
+        # subprocess holding a reference to lock1 is still alive, so this call
+        # should succeed
+        _resource_unlink(name1, rtype)
+        p.terminate()
+        p.wait()
+        time.sleep(2.0)
+        with pytest.raises(OSError) as ctx:
+            _resource_unlink(name2, rtype)
+        # docs say it should be ENOENT, but OSX seems to give EINVAL
+        assert ctx.value.errno in (errno.ENOENT, errno.EINVAL)
+        err = p.stderr.read().decode('utf-8')
+        p.stderr.close()
+        # TODO: output per type?
+        expected = ('resource_tracker: There appear to be 2 leaked '
+                    'resources')
+        assert re.search(expected, err) is not None
+
+        # lock1 is still registered, but was destroyed externally: the tracker
+        # is expected to complain.
+        expected = ("resource_tracker: %s: (OSError\\(%d|"
+                    "FileNotFoundError)" % (name1, errno.ENOENT))
+        assert re.search(expected, err) is not None
+
+    def check_resource_tracker_death(self, signum, should_die):
+        # bpo-31310: if the semaphore tracker process has died, it should
+        # be restarted implicitly.
+        from loky.backend.resource_tracker import _resource_tracker
+        pid = _resource_tracker._pid
+        if pid is not None:
+            os.kill(pid, signal.SIGKILL)
+            os.waitpid(pid, 0)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            _resource_tracker.ensure_running()
+            # in python2.7, the race condition described in bpo-33613 still
+            # exists.
+            # TODO: prevent race condition in python2.7?
+            time.sleep(1.0)
+        pid = _resource_tracker._pid
+
+        os.kill(pid, signum)
+        time.sleep(1.0)  # give it time to die
+
+        ctx = get_context("loky")
+        with warnings.catch_warnings(record=True) as all_warn:
+            warnings.simplefilter("always")
+
+            # remove unrelated MacOS warning messages first
+            warnings.filterwarnings(
+                "ignore", message='semaphore are broken on OSX')
+
+            sem = ctx.Semaphore()
+            sem.acquire()
+            sem.release()
+            wr = weakref.ref(sem)
+            # ensure `sem` gets collected, which triggers communication with
+            # the semaphore tracker
+            del sem
+            gc.collect()
+            assert wr() is None
+            if should_die:
+                assert len(all_warn) == 1
+                the_warn = all_warn[0]
+                assert issubclass(the_warn.category, UserWarning)
+                assert "resource_tracker: process died" in str(
+                    the_warn.message)
+            else:
+                print(all_warn)
+                assert len(all_warn) == 0
```

Comment:
We should get rid of debug prints in test. The `-l` flag of pytest should be informative enough in case of assertion failure.